add label to wrapper widget class

rkanavath opened the pull request:
## Description
Part of pull request https://github.com/qgis/QGIS/pull/6272 to allows widgetwrapper to access label.

https://github.com/qgis/QGIS/pull/6272#issuecomment-372982492

There is also a future possibility to create a label in widgetwrapper class rather than ParametersPanel via WidgetWrapper.createWidget()
This allows to manage only list of wrappers and in AlgorithmDialog.  Right now there is self.wrappers (to hold list of wrapper)
self.labels (to hold list of labels)

@nyalldawson, can you confirm the above case using createWidget() ?

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

nyalldawson commented:
+1 from me.

@alexbruy your thoughts?

nyalldawson commented:
> There is also a future possibility to create a label in widgetwrapper class rather than ParametersPanel via WidgetWrapper.createWidget()
This allows to manage only list of wrappers and in AlgorithmDialog. Right now there is self.wrappers (to hold list of wrapper)
self.labels (to hold list of labels)

If you mean adding a createLabel method to WidgetWrapper, then I'm +1 to this too. Sounds like a good change, and could allow removal of this hardcoded workaround: https://github.com/qgis/QGIS/blob/master/python/plugins/processing/gui/ParametersPanel.py#L144

I'm not really sure what the intended use of ParametersPanel.labels is anyway. Looks like it's written but never used.

Commit:
[processing] create and manage label in WidgetWrapper

ModelerParametersDialog and ParametersPanel have to keep list of
wrappers only. widget and label( if needed) are created through
WidgetWrapper.createLabel()

rkanavath commented:
done. pr added with  a QEP
https://github.com/qgis/QGIS-Enhancement-Proposals/issues/117

nyalldawson commented on python/plugins/processing/gui/ParametersPanel.py:
```diff
@@ -141,21 +132,21 @@ def initWidgets(self):
-                        # label.setToolTip(tooltip)
-                        self.labels[param.name()] = label
-
+                    if wrapper.label:
```
if wrapper.label is not None:

nyalldawson commented on python/plugins/processing/gui/wrappers.py:
```diff
@@ -240,6 +259,11 @@ def value(self):
 
 class BooleanWidgetWrapper(WidgetWrapper):
 
+    def __init__(self, param, dialog, row=0, col=0, **kwargs):
```
Instead of this, could you override createLabel() with something like:

    def createLabel(self):
        if self.dialogType == DIALOG_STANDARD:
            return None
        else:
            return super().createLabel()

nyalldawson commented on python/plugins/processing/gui/wrappers.py:
```diff
@@ -172,6 +174,23 @@ def comboValue(self, validator=None, combobox=None):
+        label = QLabel(desc)
+        #check this with nyall or alex, if we can use parameter key as tooltip.
+        #this can be helpful to remember commandline key for some users.
+        # label.setToolTip(tooltip)
```
yes - use:

    label.setToolTip(self.param.toolTip())

nyalldawson commented on python/plugins/processing/modeler/ModelerParametersDialog.py:
```diff
@@ -153,14 +142,14 @@ def setupUi(self):
                 self.valueItems[param.name()] = widget
                 tooltip = param.description()
-                label.setToolTip(tooltip)
+                #check if this can be moved to WidgetWrapper.createLabel(self)
```
Should be removable if changes mentioned for createLabel are made

nyalldawson commented on python/plugins/processing/modeler/ModelerParametersDialog.py:
```diff
@@ -153,14 +142,14 @@ def setupUi(self):
                 tooltip = param.description()
-                label.setToolTip(tooltip)
+                #check if this can be moved to WidgetWrapper.createLabel(self)
+                if wrapper.label:
```
if wrapper.label is not None:

rkanavath commented:
Many thanks for review. I had add all your fixes.

nyalldawson commented:
@alexbruy gentle ping - are you ok with this approach?
